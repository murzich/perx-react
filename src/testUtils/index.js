// eslint-disable-next-line import/prefer-default-export
export { default as renderWithRedux } from './renderWithRedux';
