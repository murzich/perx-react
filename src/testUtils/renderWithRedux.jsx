// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from '../state/reducer';

export default function renderWithRedux(
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store,
  };
}

export const rerenderWithRedux = rerender => (
  ui,
  { initialState, store = createStore(reducer, initialState) } = {},
) => ({
  ...rerender(<Provider store={store}>{ui}</Provider>),
  store,
});
