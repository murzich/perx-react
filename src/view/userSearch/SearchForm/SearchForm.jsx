import PropTypes from 'prop-types';
import React, { useContext } from 'react';
import { compose } from 'redux';
import { Button, Input, Message } from 'semantic-ui-react';
import { combinedLoading } from '../../../state/selectors';
import { withErrorMessage, withLoadingState } from '../../hocs';
import UserSearchContext from '../UserSearchContext';

const fieldName = 'user';

const SearchForm = ({ loading = false, errorMessage = null }) => {
  const { setUserName } = useContext(UserSearchContext);

  const handleSubmit = (event) => {
    event.preventDefault();
    const { value } = event.target.elements[fieldName];

    setUserName(value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <Input
        icon="github"
        iconPosition="left"
        type="search"
        name={fieldName}
        width={6}
        placeholder="Type user nameâ€¦"
        action={<Button type="submit">search</Button>}
        loading={loading}
        aria-label="Search by username"
        error={!!errorMessage}
      />

      {errorMessage && <Message warning header={errorMessage} />}
    </form>
  );
};

SearchForm.propTypes = {
  loading: PropTypes.bool,
  errorMessage: PropTypes.string,
};

export default compose(
  withErrorMessage,
  withLoadingState(combinedLoading),
)(SearchForm);
