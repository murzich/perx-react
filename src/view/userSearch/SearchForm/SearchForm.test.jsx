import { cleanup } from '@testing-library/react';
import React from 'react';
import { renderWithRedux } from '../../../testUtils';
import { UserSearchContext } from '../index';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  afterEach(cleanup);

  const testUser = 'testUser';

  it('should match snapshot without props', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <SearchForm />
      </UserSearchContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match snapshot with error', () => {
    const { asFragment } = renderWithRedux(
      <UserSearchContext.Provider value={testUser}>
        <SearchForm />
      </UserSearchContext.Provider>,
      {
        initialState: {
          ui: {
            error: 'testMessage',
          },
        },
      },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
