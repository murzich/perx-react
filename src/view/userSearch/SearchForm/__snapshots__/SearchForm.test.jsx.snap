// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match snapshot with error 1`] = `
<DocumentFragment>
  <form>
    <div
      class="ui error action left icon input"
      width="6"
    >
      <input
        aria-label="Search by username"
        name="user"
        placeholder="Type user name…"
        type="search"
      />
      <i
        aria-hidden="true"
        class="github icon"
      />
      <button
        class="ui button"
        type="submit"
      >
        search
      </button>
    </div>
    <div
      class="ui warning message"
    >
      <div
        class="content"
      >
        <div
          class="header"
        >
          testMessage
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`SearchForm should match snapshot without props 1`] = `
<DocumentFragment>
  <form>
    <div
      class="ui action left icon input"
      width="6"
    >
      <input
        aria-label="Search by username"
        name="user"
        placeholder="Type user name…"
        type="search"
      />
      <i
        aria-hidden="true"
        class="github icon"
      />
      <button
        class="ui button"
        type="submit"
      >
        search
      </button>
    </div>
  </form>
</DocumentFragment>
`;
